src/crypto.js

```diff
@@ -0,0 +1,64 @@
+import libsodium from 'libsodium-wrappers';
+
+/** Class that holds all cryptographic functionality. */
+export class Crypto {
+  /** Create an instance of Crypto. */
+  constructor() {
+    return (async () => {
+      await libsodium.ready;
+      this._sodium = libsodium;
+
+      return this;
+    })();
+  }
+
+  /**
+    * @param {Uint8Array|String} data - Plaintext to be encrypted.
+    * @param {Uint8Array} pk - Public key used for symmetric encryption.
+    * @return {JSON} Object containing ciphertext and nonce in Uint8Array format.
+    */
+  encryptWithKey(data, pk) {
+    const nonce = this._sodium.randombytes_buf(this._sodium.crypto_secretbox_NONCEBYTES);
+    const cipherData = this._sodium.crypto_secretbox_easy(data, nonce, pk);
+
+    // Remove plaintext value from memory
+    pk = null;
+
+    return {
+      ciphertext: cipherData,
+      nonce: nonce,
+    };
+  }
+
+  /**
+    * @param {Uint8Array} cipherData - Ciphertext to be decrypted.
+    * @param {Uint8Array} pk - Public key used for symmetric encryption.
+    * @param {Uint8Array} nonce - To be used one time only.
+    * @return {Uint8Array} Plaintext value.
+    */
+  decryptWithKey(cipherData, pk, nonce) {
+    const plaintextData = this._sodium.crypto_secretbox_open_easy(cipherData, nonce, pk);
+
+    // Remove plaintext value from memory
+    pk = null;
+
+    return plaintextData;
+  }
+
+  /**
+    * @param {Uint8Array} cipherData - Ciphertext to be decrypted.
+    * @param {Uint8Array} ck - Private key used for symmetric encryption.
+    * @param {Uint8Array} nonce - To be used one time only.
+    * @return {JSON} Cipher object with Bas64 encoded values.
+    */
+  fileCipherObject(cipherData, ck, nonce) {
+    if (cipherData.constructor !== Uint8Array || ck.constructor !== Uint8Array || nonce.constructor !== Uint8Array) {
```

Comment:
Yes

Reply:
ok, so there's no support on ES6 or ES7 for static type checking.
To check types on these specifications we can either do `.constructor ===` or `instanceof`.

Reply:
But why use pure JS and not Typescript for this library? There's a lot of added value with TS.

Reply:
I agree TypeScript has some very nice features, but that also comes with a price. I considered it at first, but then I thought it wasn't worth the investment, since es6 adds all the functionality that I needed, except static type validation, which is used in 1 method so far. And that is addressed by the unit tests added to that method.

One drawback on using TS is that the implementation would require considerably extra time. Adding types everywhere is not trivial. More when it comes to the implementation of a library that was not written in Typescript: libsodiumJS uses pure javascript. And to add type declarations to a library that wasn't written in TS, we'd need to write an ambient module.

I didn't think it's worth the investment, or at least for this first version. But I'm open to discussion, I'll reach out to you.